src/thor/dijkstras.cc

```diff
@@ -0,0 +1,1040 @@
+      } else {
+        // No matching departures found for this edge
+        continue;
+      }
+    } // This is not a transit edge
+    else {
+      // If current mode is public transit we should only connect to
+      // transit connection edges or transit edges
+      if (mode_ == TravelMode::kPublicTransit) {
+        // Disembark from transit and reset walking distance
+        mode_ = TravelMode::kPedestrian;
+        walking_distance = 0;
+        mode_change = true;
+      }
+
+      // Regular edge - use the appropriate costing and check if access
+      // is allowed. If mode is pedestrian this will validate walking
+      // distance has not been exceeded.
+      if (!mode_costing[static_cast<uint32_t>(mode_)]->Allowed(directededge, pred, tile, edgeid, 0, 0,
+                                                               has_time_restrictions)) {
+        continue;
+      }
+
+      Cost c = mode_costing[static_cast<uint32_t>(mode_)]->EdgeCost(directededge, tile);
+      c.cost *= mode_costing[static_cast<uint32_t>(mode_)]->GetModeFactor();
+      newcost += c;
+
+      // Add to walking distance
+      if (mode_ == TravelMode::kPedestrian) {
+        walking_distance += directededge->length();
+
+        // Prevent going from one egress connection directly to another
+        // at a transit stop - this is like entering a station and exiting
+        // without getting on transit
+        if (nodeinfo->type() == NodeType::kTransitEgress && pred.use() == Use::kEgressConnection &&
+            directededge->use() == Use::kEgressConnection) {
+          continue;
+        }
+      }
+    }
+
+    // Add mode change cost or edge transition cost from the costing model
+    Cost transition_cost{};
+    if (mode_change) {
+      // TODO: make mode change cost configurable. No cost for entering
+      // a transit line (assume the wait time is the cost)
+      // transition_cost = { 10.0f, 10.0f };
+    } else {
+      transition_cost =
+          mode_costing[static_cast<uint32_t>(mode_)]->TransitionCost(directededge, nodeinfo, pred);
+    }
+    newcost += transition_cost;
+
+    // Prohibit entering the same station as the prior.
+    if (directededge->use() == Use::kTransitConnection &&
+        directededge->endnode() == pred.prior_stopid()) {
+      continue;
+    }
+
+    // Test if exceeding maximum transfer walking distance
+    if (directededge->use() == Use::kTransitConnection && pred.prior_stopid().Is_Valid() &&
+        walking_distance > max_transfer_distance_) {
+      continue;
+    }
+
+    // TODO: need to put this in the base class and let isochrones implement it
+    // TODO: can we cram it into the expansion recommendation?
+    // Continue if the time interval has been met. This bus or rail line goes beyond the max
+    // but need to consider others so we just continue here.
+    /*
+    if (newcost.secs > max_seconds_) {
+      continue;
+    }
+    */
```

Comment:
This is the only thing left to do on this PR. can we cram this into the recommendation callback thing? or do we need to add a new callback which has semantics like:

`bool ShouldIgnoreEdgeLabel(const directededge* e, const cost& newcost) const;`

Reply:
Function wise, this could be handled via the existing callback `ShouldExpand` in `ComputeMultiModal` after `pred` has been popped from the priority queue, yes?

The benefit of doing it here would be that it never would be added to the priority queue which is a performance win.

Perhaps another callback that acts on `DirectedEdge` + `Cost`? Or could we create an `EdgeLabel` here and reuse the existing `ShouldExpand` function? Reusing the existing callback may be nicer for readers and users of the code unless there's a performance hit to creating the label.